src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/FrameHandling/FrameIterator.cs

```diff
@@ -0,0 +1,160 @@
+
+            // TransitionFrame type frames
+            case FrameType.FramedMethodFrame:
+            case FrameType.CLRToCOMMethodFrame:
+            case FrameType.PInvokeCalliFrame:
+            case FrameType.PrestubMethodFrame:
+            case FrameType.StubDispatchFrame:
+            case FrameType.CallCountingHelperFrame:
+            case FrameType.ExternalMethodFrame:
+            case FrameType.DynamicHelperFrame:
+                Data.FramedMethodFrame framedMethodFrame = target.ProcessedData.GetOrAdd<Data.FramedMethodFrame>(CurrentFrame.Address);
+                Data.TransitionBlock transitionBlock = target.ProcessedData.GetOrAdd<Data.TransitionBlock>(framedMethodFrame.TransitionBlockPtr);
+                if (target.GetTypeInfo(DataType.TransitionBlock).Size is not uint transitionBlockSize)
+                {
+                    throw new InvalidOperationException("TransitionBlock size is not set");
+                }
+                return GetFrameHandler(context).HandleTransitionFrame(framedMethodFrame, transitionBlock, transitionBlockSize);
+
+            case FrameType.FuncEvalFrame:
+                Data.FuncEvalFrame funcEvalFrame = target.ProcessedData.GetOrAdd<Data.FuncEvalFrame>(CurrentFrame.Address);
+                Data.DebuggerEval debuggerEval = target.ProcessedData.GetOrAdd<Data.DebuggerEval>(funcEvalFrame.DebuggerEvalPtr);
+                return GetFrameHandler(context).HandleFuncEvalFrame(funcEvalFrame, debuggerEval);
+
+            // ResumableFrame type frames
+            case FrameType.ResumableFrame:
+            case FrameType.RedirectedThreadFrame:
+                Data.ResumableFrame resumableFrame = target.ProcessedData.GetOrAdd<Data.ResumableFrame>(CurrentFrame.Address);
+                return GetFrameHandler(context).HandleResumableFrame(resumableFrame);
+
+            case FrameType.FaultingExceptionFrame:
+                Data.FaultingExceptionFrame faultingExceptionFrame = target.ProcessedData.GetOrAdd<Data.FaultingExceptionFrame>(CurrentFrame.Address);
+                return GetFrameHandler(context).HandleFaultingExceptionFrame(faultingExceptionFrame);
+
+            case FrameType.HijackFrame:
+                Data.HijackFrame hijackFrame = target.ProcessedData.GetOrAdd<Data.HijackFrame>(CurrentFrame.Address);
+                return GetFrameHandler(context).HandleHijackFrame(hijackFrame);
+            default:
+                return false;
+        }
+    }
+
+    public bool IsInlineCallFrameWithActiveCall()
+    {
+        if (GetFrameType(CurrentFrame) != FrameType.InlinedCallFrame)
+        {
+            return false;
+        }
+        Data.InlinedCallFrame inlinedCallFrame = target.ProcessedData.GetOrAdd<Data.InlinedCallFrame>(currentFramePointer);
+        return inlinedCallFrame.CallerReturnAddress != 0;
+    }
+
+    private FrameType GetFrameType(Data.Frame frame)
+    {
+        foreach (FrameType frameType in Enum.GetValues<FrameType>())
```

Comment:
If I'm not mistaken you could use Enum.GetNames instead and avoid the ToString call in the inner loop. But I'm not sure it matters